core/src/main/java/org/apache/iceberg/expressions/ExpressionParser.java

```diff
@@ -229,6 +242,44 @@ private void unboundLiteral(Object object) throws IOException {
       }
     }
 
+    private void geospatialBoundingBox(BoundingBox value) throws IOException {
```

Comment:
maybe move the bounding box code to a separate package private `BoundingBoxParser` class?